src/deserialize/parse.rs

```diff
@@ -197,34 +197,79 @@ fn read_discard(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Opt
 }
 
 fn read_number(n: char, chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Edn, Error> {
+    let delimiters = [',', ']', '}', ')', ';'];
     let i = chars
         .clone()
         .next()
         .ok_or_else(|| Error::ParseEdn("Could not identify symbol index".to_string()))?
         .0;
     let c_len = chars
         .clone()
-        .take_while(|c| {
-            c.1.is_numeric()
-                || c.1 == '.'
-                || c.1 == '/'
-                || c.1 == 'e'
-                || c.1 == 'E'
-                || c.1 == '+'
-                || c.1 == '-'
-        })
+        .take_while(|(_, c)| !c.is_whitespace() && !delimiters.contains(c))
         .count();
-    let mut number = String::new();
-    let string = chars.take(c_len).map(|c| c.1).collect::<String>();
-    number.push(n);
-    number.push_str(&string);
+    let (number, radix) = {
+        let mut number = String::new();
+        number.push(n);
+        for (_, c) in chars.take(c_len) {
+            number.push(c)
+        }
+        if number.to_lowercase().starts_with("0x") {
```

Comment:
I would create a new parser for stuff staring with `0x` and not reuse this one

Reply:
I think this should live here, but am open to suggestions. 

I think moving chars (all about parse.rs) as a &str would allow for more powerful pattern matching without having to mutate the iterator. For example it might be nice to be able to match against `-0x...` and `0x...` here
https://github.com/naomijub/edn-rs/blob/8353fc4cbfb1938849ffed0a4a53951b95706f24/src/deserialize/parse.rs#L41

This would also likely be a huge performance gain, because there wouldn't need to be heap allocating clones just for reading the index or lengths of things, just to be tossed away.

I think this goes beyond the scope of non-dec and should be done independently (and probably after #104)

Reply:
Also thought about adding it to here `pub(crate) fn parse_edn`

Reply:
I don't think this should be cleaned up without #105.

I'm biased because I specifically need non-doc, but also I think it might end up being here anyway. The pattern of needing to escape out 0x Nr, with minus symbols, will need an allocation and should probably be together anyway. I think #105 will make things obvious and at the very least a working implementation with test coverage should happen first.